### PR TITLE
Makes Space Hotel Great Again

### PR DESCRIPTION
**Pull request by Tastyfish**
This completely reworks the space hotel to hopefully be more interesting.
- Each room is unique, although some are more interesting than others.
- Furthermore, the rooms are randomly assigned (using map templates), and there are more possible rooms than slots on the map.
- Some rooms are _a lot_ more interesting than others.
- Probably exciting exploits to discover and be squashed!
- Other things!

:cl:
rscadd: There's essentially a completely new space hotel map! Explorers rejoice!
/:cl:

### CODE REVIEW

**Comment by tigercat2000** [comment]
#4700 
:ok_hand:

**Comment by taukausanake** [comment]
Oh wow this actually looks like something and totally not a changelog PR. Cool.

Someone explain to me what the Space Hotel is or was exactly. How does it and how did it work before? Spare no detail as I am very interested in this

**Comment by Tastyfish** [comment]
It's currently an away mission that has some dead alium stuff, some crossbows, bloody candles, and bedsheets. The new one is better. :wink: I slightly want the actual content to be a surprise, although it does have SNPC staff and things that will hopefully at least be cool the first few times it's explored for people that like the exploration angle of away missions.

**Comment by taukausanake** [comment]
Wait, was this the map that had xeno eggs in it and would return to the station infected so they could be aliens? Or the one with a whole lot of deads around and some syndicate in a vault out there? I know which one this is I think

**Comment by Tastyfish** [comment]
The one with carpets. Yeah, it used to have an egg.

**Comment by Keilden** [comment]
There should be a room with a morgue slab, plasmaman suit and a husk.

**Comment by Fox-McCloud** [comment]
As someone who got to super-early test this :+1: 

It's pretty darn good.

**Comment by Taaketa** [comment]
I think the important question here is... Will there be a wall and whose gonna pay for it?

**Comment by Tastyfish** [comment]
Yes.

**Comment by TheDZD** [comment]
This has conflicts.

**Comment by Tastyfish** [comment]
I think I'm finished fiddling with this.

File: code/modules/awaymissions/mission_code/spacehotel.dm [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+	if(radio)
+		qdel(radio)
+		radio = null
+
+	return ..()
+
+// to check a person into a room; no financial stuff; returns the keycard
+/obj/effect/hotel_controller/proc/checkin(roomid, mob/living/carbon/occupant, obj/item/weapon/card/id/id)
+	if(!istype(occupant))
+		world << "no occupant"
```
Comment (Crazylemon64) [inline code/modules/awaymissions/mission_code/spacehotel.dm]: This appears to be left-in debugging text hidden by goonchat

Comment (Tastyfish) [inline code/modules/awaymissions/mission_code/spacehotel.dm]: oop

**Pull request merged**